Content.Server/Implants/ImplanterSystem.cs

```diff
@@ -0,0 +1,185 @@
+﻿using System.Threading;
+using Content.Server.DoAfter;
+using Content.Server.Guardian;
+using Content.Server.Popups;
+using Content.Shared.Hands;
+using Content.Shared.IdentityManagement;
+using Content.Shared.Implants;
+using Content.Shared.Implants.Components;
+using Content.Shared.Interaction;
+using Content.Shared.MobState.Components;
+using Content.Shared.Popups;
+using Robust.Shared.Containers;
+using Robust.Shared.GameStates;
+using Robust.Shared.Player;
+
+namespace Content.Server.Implants;
+
+public sealed class ImplanterSystem : SharedImplanterSystem
+{
+    [Dependency] private readonly PopupSystem _popup = default!;
+    [Dependency] private readonly DoAfterSystem _doAfter = default!;
+    [Dependency] private readonly SharedContainerSystem _container = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<ImplanterComponent, HandDeselectedEvent>(OnHandDeselect);
+        SubscribeLocalEvent<ImplanterComponent, AfterInteractEvent>(OnImplanterAfterInteract);
+        SubscribeLocalEvent<ImplanterComponent, ComponentGetState>(OnImplanterGetState);
+
+        SubscribeLocalEvent<ImplanterComponent, ImplanterImplantCompleteEvent>(OnImplantAttemptSuccess);
+        SubscribeLocalEvent<ImplanterComponent, ImplanterDrawCompleteEvent>(OnDrawAttemptSuccess);
+        SubscribeLocalEvent<ImplanterComponent, ImplanterCancelledEvent>(OnImplantAttemptFail);
+    }
+
+    private void OnImplanterAfterInteract(EntityUid uid, ImplanterComponent component, AfterInteractEvent args)
+    {
+        if (args.Target == null || !args.CanReach || args.Handled)
+            return;
+
+        if (component.CancelToken != null)
+        {
+            args.Handled = true;
+            return;
+        }
+
+        //Simplemobs and regular mobs should be injectable, but only regular mobs have mind.
+        //So just don't implant/draw anything that isn't living or is a guardian
+        //TODO: Rework a bit when surgery is in to work with implant cases
+        if (!HasComp<MobStateComponent>(args.Target.Value) || HasComp<GuardianComponent>(args.Target.Value))
+            return;
+
+        //TODO: Rework when surgery is in for implant cases
+        if (component.CurrentMode == ImplanterToggleMode.Draw && !component.ImplantOnly)
+        {
+            TryDraw(component, args.User, args.Target.Value, uid);
+        }
+        else
+        {
+            //Implant self instantly, otherwise try to inject the target.
+            if (args.User == args.Target)
+                Implant(uid, args.Target.Value, component);
```

Comment:
Think it should still have a delay but a shorter one.

Reply:
I can do 0.1 if you want, but Hyposprays also don't have a self inject timer.

Reply:
Hyposprays are also pretty busted as people just spam click it.